llama_index/readers/file/image_caption_parser.py

```diff
@@ -18,57 +19,59 @@ def __init__(
         prompt: Optional[str] = None,
     ):
         """Init params."""
-        self._parser_config = parser_config
-        self._keep_image = keep_image
-        self._prompt = prompt
-
-    def _init_parser(self) -> Dict:
-        """Init parser."""
-        try:
-            import torch  # noqa: F401
-        except ImportError:
-            raise ImportError(
-                "install pytorch to use the model: " "`pip install torch`"
-            )
-        try:
-            from transformers import BlipProcessor, BlipForConditionalGeneration
-        except ImportError:
-            raise ImportError(
-                "transformers is required for using BLIP model: "
-                "`pip install transformers`"
-            )
-        try:
-            import sentencepiece  # noqa: F401
-        except ImportError:
-            raise ImportError(
-                "sentencepiece is required for using BLIP model: "
-                "`pip install sentencepiece`"
+        if parser_config is None:
+            """Init parser."""
+            try:
+                import torch  # noqa: F401
+            except ImportError:
+                raise ImportError(
+                    "install pytorch to use the model: " "`pip install torch`"
```

Comment:
Can we combine all these? Would be annoying to install each of these one at a time and hit a new error on every retry

```
Please install extra dependencies that are required for the ImageCaptionReader: 
`pip install torch transformers sentencepiece Pillow`
```